### PR TITLE
Simplify CI

### PR DESCRIPTION
**Pull request by nirs**
Split CI to multiple workflow so we can run qemu and lima integration tests
without conflicts on all platforms.

- build: build on multiple version of macOS
- lima: lima integration tests
- qemu: lima integration tests
- release: already exist

Try to run integration tests on all platforms to see what works now.  We  
probably want to minimize the number of platform since the macOS runners
are very flaky.

The lima workflow is flaky. Sometimes all 3 platforms succeed, but 
usually one gets stuck and cancelled after many minutes.

Notes:                                                                                                                                                                       
- Workflow names unified to Title case. This is more consistent with
  github UI and standard workflow names.

Workarounds:
- on macos-15-large brew is not updated and get lima 0.23.2. Fixed by
  running brew update.

### CODE REVIEW

**Comment by nirs** [comment]
Moving to draft to test standard qemu

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          - macos-14
+          - macos-15
+    runs-on: ${{ matrix.platform }}
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 1
+      - name: Build
+        run: make
```
Comment (AkihiroSuda) [inline .github/workflows/build.yaml:30]: `make test.repro` should be tested

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # Forked from https://github.com/containerd/nerdctl/blob/v0.8.1/.github/workflows/release.yml
 # Apache License 2.0
 
-name: Release
+name: release
```
Comment (AkihiroSuda) [inline .github/workflows/release.yaml:5]: ?

File: .github/workflows/qemu.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        # macos-13-large is used as macos-13 seems too flaky.
+        # macos-14 (ARM) and later cannot be used for the most part of the job
+        # due to the lack of the support for nested virt.
+        platform:
+          - macos-13-large
+          - macos-14-large
+          - macos-15-large
```
Comment (AkihiroSuda) [inline .github/workflows/qemu.yaml:25]: QEMU test is quite lightweight and should be executable on ARM Mac with QEMU TCG mode

Comment (AkihiroSuda) [reply .github/workflows/qemu.yaml:25]: So, you can just merge qemu.yaml to build.yaml

File: .github/workflows/build.yaml (hunk 1 above)
Comment (nirs) [reply .github/workflows/build.yaml:30]: I think I copied it from #63. You want to replace with `make` with `make test.repro`?

File: .github/workflows/release.yaml (hunk 2 above)
Comment (nirs) [reply .github/workflows/release.yaml:5]: For consistency with other names.

File: .github/workflows/qemu.yaml (hunk 3 above)
Comment (nirs) [reply .github/workflows/qemu.yaml:25]: I can switch to arm, but I don't want to mix integration tests with trivial build. This test is likely to break in the future and will requires workaround, lets keep it separate.

Comment (nirs) [reply .github/workflows/qemu.yaml]: Updated to run with arm64 machines, kept as separate workflow for easier future maintainance.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (nirs) [reply .github/workflows/build.yaml:30]: The release workflow run `make test.repro` why test it here again? If we want to test this on all architectures we can add more platforms to the release matrix.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**